tt_metal/hw/firmware/src/brisc.cc

```diff
@@ -425,7 +437,12 @@ int main() {
                 RECORD_STACK_USAGE();
             } else {
                 // This was not initialized in kernel_init
-                noc_local_state_init(noc_index);
+                if (noc_mode == DM_DEDICATED_NOC) {
```

Comment:
hmm.  this is the rare case - BRISC w/o a kernel.  I'm a little concerned about FW code bloat.  I wonder if:
noc_mode = DM_DEDICATED_NOC;
noc_local_state_init(noc_index);
would be better